--- a/vidcutter/libs/mpvwidget.py	2017-12-15 09:54:25.000000000 -0600
+++ b/vidcutter/libs/mpvwidget-1.py	2017-12-26 11:25:14.629090000 -0600
@@ -6,6 +6,11 @@
 import os
 import sys
 
+# this is required for Ubuntu which seems to have a broken PyQt5 OpenGL implementation
+# so we use PyOpenGL instead for the GL context
+# noinspection PyUnresolvedReferences
+from OpenGL import GL
+
 from PyQt5.QtCore import pyqtSignal, pyqtSlot, Qt, QEvent, QTimer
 from PyQt5.QtGui import QKeyEvent, QMouseEvent, QWheelEvent
 from PyQt5.QtOpenGL import QGLContext
@@ -156,7 +161,7 @@
     def setLogLevel(self, loglevel):
         self.mpv.set_log_level(loglevel)
 
-    def showText(self, msg: str, duration: int=5, level: int=0):
+    def showText(self, msg: str, duration: int = 5, level: int = 0):
         self.mpv.command('show-text', msg, duration * 1000, level)
 
     def play(self, filepath) -> None:
@@ -181,10 +186,10 @@
     def volume(self, vol: int) -> None:
         self.mpv.set_property('volume', vol)
 
-    def codec(self, stream: str='video') -> str:
-        return self.mpv.get_property('{}-codec'.format(stream))
+    def codec(self, stream: str = 'video') -> str:
+        return self.mpv.get_property('audio-codec' if stream == 'audio' else 'video-codec')
 
-    def format(self, stream: str='video') -> str:
+    def format(self, stream: str = 'video') -> str:
         return self.mpv.get_property('audio-codec-name' if stream == 'audio' else 'video-format')
 
     def property(self, prop: str):
